boot.go

```diff
@@ -43,10 +47,51 @@ func main() {
 					pkglog.Err("getting config for %s [%s]", serverConfAppName, err)
 					os.Exit(1)
 				}
-				pkglog.Info("starting fetcher on port %d", cnf.FetcherPort)
-				go fetcher.Serve(cnf.FetcherPort)
-				pkglog.Info("starting SSH server on %s:%d", cnf.SSHHostIP, cnf.SSHHostPort)
-				os.Exit(pkg.Run(cnf.SSHHostIP, cnf.SSHHostPort, "boot"))
+				circ := sshd.NewCircuit()
+
+				s3Client, err := storage.GetClient(cnf.HealthSrvTestStorageRegion)
+				if err != nil {
+					log.Printf("Error getting s3 client (%s)", err)
+					os.Exit(1)
+				}
+				kubeClient, err := kcl.NewInCluster()
+				if err != nil {
+					log.Printf("Error getting kubernetes client [%s]", err)
+					os.Exit(1)
+				}
+				log.Printf("Starting health check server on port %d", cnf.HealthSrvPort)
+				healthSrvCh := make(chan error)
+				go func() {
+					if err := healthsrv.Start(cnf.HealthSrvPort, kubeClient.Namespaces(), s3Client, circ); err != nil {
+						healthSrvCh <- err
```

Comment:
Well, the goal here is to check that the k8s API is accessible. Given that the builder has started, we can assume that the namespace under which it runs (also specified in `POD_NAMESPACE`) exists. The call that the health server makes is to just get a list of namespaces, and the goal there is only to see if the master is accessible.

I think @aledbf had an idea for a cheaper method of determining whether the API server was accessible, but if you have ideas I'd like to hear them too.